src/hotspot/share/jeandle/jeandleAbstractInterpreter.cpp

```diff
@@ -101,34 +101,34 @@ bool JeandleVMState::update_phi_nodes(JeandleVMState* income_jvm, llvm::BasicBlo
     return false;
   }
 
-  llvm::SmallVector<llvm::Value*>& income_locals = income_jvm->_locals;
-  llvm::SmallVector<llvm::Value*>& income_stack = income_jvm->_stack;
+  llvm::SmallVector<TypedValue>& income_locals = income_jvm->_locals;
+  llvm::SmallVector<TypedValue>& income_stack = income_jvm->_stack;
 
   // Create phi nodes for locals.
   for (size_t i = 0; i < _locals.size(); i++) {
-    if (_locals[i] == nullptr) {
+    if (_locals[i].is_null()) {
       continue;
     }
 
-    llvm::PHINode* phi_node = llvm::cast<llvm::PHINode>(_locals[i]);
+    llvm::PHINode* phi_node = llvm::cast<llvm::PHINode>(_locals[i].value());
 
-    if (income_locals[i] == nullptr || phi_node->getType() != income_locals[i]->getType()) {
+    if (income_locals[i].is_null() || phi_node->getType() != income_locals[i].value()->getType()) {
```

Comment:
Also use the basic type, not the llvm type. The previous implementation was not rigorous.